pkg/loop/service.go

```diff
@@ -212,11 +212,43 @@ func (s *pluginService[P, S]) closeClient() (err error) {
 	return
 }
 
-func (s *pluginService[P, S]) wait(ctx context.Context) error {
+func (s *PluginService[P, S]) Wait(ctx context.Context) error {
 	select {
 	case <-ctx.Done():
 		return context.Cause(ctx)
 	case <-s.serviceCh:
 		return nil
+	case <-s.stopCh:
+		return nil
+	}
+}
+
+// XXXTestHook returns a TestPluginService.
+// It must only be called once, and before Start.
+func (s *PluginService[P, S]) XXXTestHook() TestPluginService[P, S] {
+	s.testInterrupt = make(chan func(*PluginService[P, S]))
+	return s.testInterrupt
+}
+
+// TestPluginService supports Killing & Resetting a running *pluginService.
+type TestPluginService[P grpcPlugin, S services.Service] chan<- func(*PluginService[P, S])
```

Comment:
Why export a test type?

Reply:
It's used by the `reporting_plugins` package.